Content.Shared/_RMC14/Xenonids/Parasite/SharedXenoParasiteSystem.cs

```diff
@@ -361,17 +370,102 @@ public override void Update(float frameTime)
                 if (infected.IncubationMultiplier < 1)
                     infected.BurstAt += TimeSpan.FromSeconds(1 - infected.IncubationMultiplier) * frameTime;
 
+                // Stages
+                // Percentage of how far along we out to burst time times the number of stages, truncated. You can't go back a stage once you've reached one
+                int stage = Math.Max((int) ((infected.BurstDelay - (infected.BurstAt - time)) / infected.BurstDelay * infected.FinalStage), infected.CurrentStage);
+                if (stage != infected.CurrentStage)
+                    infected.CurrentStage = stage;
+
+                // Warn on the last to final stage of a burst
+                if (!infected.DidBurstWarning && stage == infected.FinalStage - 1)
+                {
+                    _popup.PopupEntity(Loc.GetString("cm-xeno-infection-burst-soon-self"), uid, uid, PopupType.MediumCaution);
+                    _popup.PopupEntity(Loc.GetString("cm-xeno-infection-burst-soon", ("victim", uid)), uid, Filter.PvsExcept(uid), true, PopupType.MediumCaution);
+                    _jitter.DoJitter(uid, infected.JitterTime * 6, false);
+                    infected.DidBurstWarning = true;
+                    continue;
+                }
+                // Symptoms only start after the IntialSymptomStart is passed (by default, 2)
+                // And continue until burst time is reached
+                // TODO after burst time is reached, the larva is made and stage set to 6, have wait time for someone to take the larva
+                // Stage 7 should auto-burst the victim if we get nobody in stage 6
+                if (stage >= infected.FinalSymptomsStart)
+                {
+                    if (_random.Prob(infected.MajorPainChance * frameTime))
+                    {
+                        var message = Loc.GetString("cm-xeno-infection-majorpain-" + _random.Pick(new List<string> { "chest", "breathing", "heart" }));
+                        _popup.PopupEntity(message, uid, uid, PopupType.SmallCaution);
+                        if (_random.Prob(0.5f))
+                        {
+                            var ev = new VictimInfectedEmoteEvent(infected.ScreamId);
+                            RaiseLocalEvent(uid, ev);
```

Comment:
This is where you need to change it to be raised by ref with the change above.
```suggestion
                            RaiseLocalEvent(uid, ref ev);
```